src/main/java/com/epam/esm/entity/Identifiable.java

```
@@ -1,7 +1,17 @@
 package com.epam.esm.entity;
 
+/**
+ * This interface specifies that an object of a class implements this interface has an id field
+ *
+ * @author Alexey Bliznichenko
+ */
 public interface Identifiable {
```

Once again: why do we need the interface?

---

The main purpose of the interface is to mark entities. This is just interface-marker

---

@a1exanddrovich all marker interfaces have their purposes, they're used somewhere eventually. So do we need this interface?
+ 1 question for demo to discuss marker interfaces

---

As I learnt such interfaces provide runtime information about objects in order to let JVM obtain additional information about the object. For instance, let's have a look at the Identifiable interface. We could implement kinda a check that if a class does not implement the interface and we want to use the getId() method, we can throw an exception that the method we try to call is not supported.

---

@a1exanddrovich has this "kind of check" been implemented somewhere already? If not, please read about the YAGNI software development principle

---

Okay, dropped it

---

I had to return it in order to avoid code duplication